### PR TITLE
csharp cleanup

### PR DESCRIPTION
**Pull request by chriselion**
Apply a lot (but not 100%) of Rider's suggestions.

### CODE REVIEW

File: Project/Assets/ML-Agents/Examples/Soccer/Scripts/AgentSoccer.cs [hunk 1]
```diff
@@ -26,15 +26,14 @@ public enum Team
     [HideInInspector]
     public Rigidbody agentRb;
     SoccerSettings m_SoccerSettings;
-    Renderer m_AgentRenderer;
```
Comment (chriselion) [inline Project/Assets/ML-Agents/Examples/Soccer/Scripts/AgentSoccer.cs]: @andrewcoh A few unused variables from soccer. Just want to make sure there's no plans to bring these back.

File: com.unity.ml-agents/Runtime/Policy/BehaviorParameters.cs [hunk 2]
```diff
@@ -37,7 +36,7 @@ private enum BehaviorType
         string m_BehaviorName = "My Behavior";
         [HideInInspector]
         [SerializeField]
-        public int m_TeamID = 0;
+        public int m_TeamID;
```
Comment (chriselion) [inline com.unity.ml-agents/Runtime/Policy/BehaviorParameters.cs]: This and m_useChildSensors are still public with `m_` prefixes. Not sure if we should rename or make internal. `m_TeamID` seems like it needs to be mutable at runtime.

File: com.unity.ml-agents/Runtime/Policy/BehaviorParameters.cs [hunk 3]
```diff
@@ -93,13 +92,13 @@ internal IPolicy GeneratePolicy(Func<float[]> heuristic)
         }
 
         public void GiveModel(
-            string behaviorName,
+            string newBehaviorName,
```
Comment (chriselion) [inline com.unity.ml-agents/Runtime/Policy/BehaviorParameters.cs]: Argument name shadowed the `behaviorName` property

File: com.unity.ml-agents/Tests/Editor/EditModeTestInternalBrainTensorApplier.cs [hunk 4]
```diff
@@ -12,12 +11,7 @@ public class EditModeTestInternalBrainTensorApplier
     {
         class TestAgent : Agent
         {
-            public AgentAction GetAction()
```
Comment (chriselion) [inline com.unity.ml-agents/Tests/Editor/EditModeTestInternalBrainTensorApplier.cs]: Agent has a GetAction now, and this wasn't actually being used.

File: Project/Assets/ML-Agents/Examples/Soccer/Scripts/AgentSoccer.cs (hunk 1 above)
Comment (andrewcoh) [reply Project/Assets/ML-Agents/Examples/Soccer/Scripts/AgentSoccer.cs]: Ah I missed this. Good to remove.

**Pull request merged**